pkg/kube/statefulset/statefulset.go

```diff
@@ -53,13 +54,16 @@ type GetUpdateCreateDeleter interface {
 // CreateOrUpdate creates the given StatefulSet if it doesn't exist,
 // or updates it if it does.
 func CreateOrUpdate(getUpdateCreator GetUpdateCreator, sts appsv1.StatefulSet) (appsv1.StatefulSet, error) {
-	_, err := getUpdateCreator.GetStatefulSet(types.NamespacedName{Name: sts.Name, Namespace: sts.Namespace})
+	currSts, err := getUpdateCreator.GetStatefulSet(types.NamespacedName{Name: sts.Name, Namespace: sts.Namespace})
 	if err != nil {
 		if apiErrors.IsNotFound(err) {
 			return appsv1.StatefulSet{}, getUpdateCreator.CreateStatefulSet(sts)
 		}
 		return appsv1.StatefulSet{}, err
 	}
+	if equality.Semantic.DeepEqual(currSts, sts) {
```

Comment:
> This method is preferred since it focuses on semantic equality instead of memory equality.

I am not sure if I understand the difference here. Would you happen to have an example?

Reply:
Sure. 

for example, it matches an empty slice with a nil slice and the same for a map. 

In our code where we generate an internal view of a statefulset we create empty maps and slices while the api server returns them as nil (or the other way around). With above function we map due to semantic equality.